Tests/Functional/lit.cfg

```diff
@@ -114,21 +114,14 @@ else:
                 '-I', libdispatch_overlay_dir,
                 '-I', os.path.join(libdispatch_overlay_dir, 'swift'),
                 '-L', libdispatch_build_dir,
-                '-L', os.path.join(libdispatch_build_dir, 'src'),
                 '-L', os.path.join(libdispatch_build_dir, 'src', 'BlocksRuntime'),
                 '-L', os.path.join(libdispatch_build_dir, 'src', 'swift'),
             ])
 
-    config.environment['LD_LIBRARY_PATH'] = ":".join([
-      libdispatch_build_dir,
-      os.path.join(libdispatch_build_dir, 'src'),
-      foundation_dir,
-      os.path.join(foundation_dir, 'Foundation'),
-      os.path.join(foundation_dir, 'Sources', 'Foundation'),
-      os.path.join(foundation_dir, 'Sources', 'FoundationNetworking'),
-      os.path.join(foundation_dir, 'Sources', 'FoundationXML'),
-      os.path.join(foundation_dir, 'lib'),
-    ])
+            if platform.system() != 'Windows':
+                swift_exec.extend([
+                    '-Xlinker', '-rpath', '-Xlinker', libdispatch_build_dir,
+                ])
```

Comment:
Weird, the indentation seems odd, but the change itself seems like the right thing to do.